mono/utils/mono-context.c

```diff
@@ -233,6 +233,22 @@ mono_sigctx_to_monoctx (void *sigctx, MonoContext *mctx)
 	mctx->gregs [AMD64_R13] = context->R13;
 	mctx->gregs [AMD64_R14] = context->R14;
 	mctx->gregs [AMD64_R15] = context->R15;
+	memcpy (&(mctx->fregs [AMD64_XMM0]), &(context->Xmm0), sizeof (MonoContextSimdReg));
```

Comment:
This probably works and is then preferable:
```
mono_static_assert (sizeof(mctx->fregs) == (char*)(&context->Xmm15 + 1) - (char*)&context->Xmm0);
memcpy (&mctx->fregs, &context->Xmm0, sizeof (mctx->fregs));
```

Reply:
memcpy is an intrinsic and will be lowered to inlined moves using XMM for above code so don't believe we would gain much doing memcpy and we will start to take dependencies on layout as well, currently not needed. Will keep existing implementation.

Reply:
It'll likely be more optimized/smaller code, and is easier to read, and you can assert the layout, and the layout will never change, and it is the obvious layout.